src/main/java/net/snowflake/client/core/SFJsonResultSet.java

```diff
@@ -233,7 +233,7 @@ public byte[] getBytes(int columnIndex) throws SFException {
   }
 
   public Date getDate(int columnIndex) throws SFException {
-    return getDate(columnIndex, TimeZone.getDefault());
+    return getDate(columnIndex, TimeZone.getTimeZone("UTC"));
```

Comment:
I think we should take into account the parameter `JDBC_USE_SESSION_TIMEZONE` and then return the date in default timezone (when FALSE) or in session timezone (if TRUE)